src/components/CreatePost.js

```diff
@@ -0,0 +1,237 @@
+import React, {useState} from 'react';
+import {
+  Keyboard,
+  Modal,
+  StyleSheet,
+  Text,
+  TouchableWithoutFeedback,
+  View,
+} from 'react-native';
+import {APIPost} from '../utils/ApiClient';
+import {showToast} from '../utils/LocalStorage';
+import Button from './Button';
+import InputText from './InputText';
+import ErrorMessage from './ErrorMessage';
+import validate from './validator';
+import {FilePickerActionSheet} from './ActionSheet';
+import DashX from '@dashx/react-native';
+
+export const CreatePostModal = ({visible, dismissModal}) => {
+  const [newPost, setNewPost] = useState('');
+  const [requestInProgress, setRequestInProgress] = useState(false);
+  const [errorMessage, setErrorMessage] = useState({
+    newPost: false,
+  });
+  const [storePostImage, setStorePostImage] = useState();
+  const [storePostVideo, setStorePostVideo] = useState();
+  const [mediaType, setMediaType] = useState('');
+  const [showActionSheet, setShowActionSheet] = useState(false);
+
+  const validation = () => {
+    let count = 0;
+
+    for (let key in errorMessage) {
+      let validationResponse = validate(key, eval(key));

```

Comment:
@Charan017 let's move away from using `eval` for validation across all the forms we have. It's not recommended pattern to follow. Please create a ticket for it to track.

Reply:
Done